crates/xilem_web/src/interfaces.rs

```diff
@@ -91,11 +91,45 @@ where
         }
     }
 
-    // TODO should some methods extend some properties automatically,
-    // instead of overwriting the (possibly set) inner value
-    // or should there be (extra) "modifier" methods like `add_class` and/or `remove_class`
-    fn class(self, class: impl Into<Cow<'static, str>>) -> Attr<Self, T, A> {
-        self.attr("class", class.into())
+    /// Add a class to the wrapped element.
+    ///
+    /// If multiple classes are added, all will be applied to the element.
+    fn class(self, class: impl Into<Cow<'static, str>>) -> Class<Self, T, A> {
+        self.class_opt(Some(class))
+    }
+
+    /// Add an optional class to the wrapped element.
+    ///
+    /// If multiple classes are added, all will be applied to the element.
+    fn class_opt(self, class: Option<impl Into<Cow<'static, str>>>) -> Class<Self, T, A> {
```

Comment:
Not sure if we should maybe achieve this via a trait like I've experimented [here](https://github.com/Philipp-M/xilem_html-typing-tests/blob/5c831060d21c22e46b4f8ce471f2ec94e0230b60/examples/extension_traits-and-attr-map.rs#L196), which would allow adding multiple classes at once in more structured manner. 

Also as the previous `TODO` comment suggested, it would be nice to be able to remove classes as well (but I think `el.class()` should indeed be additive).

Reply:
Aah I quite like this approach. I will revise the PR to use it.